.github/workflows/dist.yml

```diff
@@ -119,10 +119,27 @@ jobs:
         with:
           name: dist
           path: dist
-      - uses: softprops/action-gh-release@v2
+      - name: Create release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          latest_release_tag=$(curl -s https://api.github.com/repos/${{ github.repository }}/releases \
+          | jq -r 'sort_by(.created_at) | last(.[]).tag_name')
+          curl -L \
+            -X POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer $GITHUB_TOKEN" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            https://api.github.com/repos/${{ github.repository }}/releases \
+            -d '{
+              "tag_name": "${{ github.ref_name }}",
+              "previous_tag_name": "${latest_release_tag}",
+              "prerelease": ${{ contains(github.ref, 'beta') || contains(github.ref, 'rc') }},
+              "generate_release_notes": true
+            }'
```

Comment:
I went through the [official docs](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release) and didn't found parameter `previous_tag_name` for **Create a release** endpoint. So it seems, even if `generate_release_notes` is set to true, the `previous_tag_name` plays no role.

The parameter `previous_tag_name` exists for [generate release notes](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#generate-release-notes-content-for-a-release) endpoint. This endpoint will return the `name` and `body` according to specified parameters, which could be then used to create a release using [create a release](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release) endpoint

Reply:
Nonetheless it works fine.

It seems that `"generate_release_notes": true` internally triggers the "generate release notes" endpoint as well, which uses `previous_tag_name`.

Reply:
I will check again.